cpp/src/arrow/compute/kernels/minmax.cc

```diff
@@ -106,7 +110,13 @@ class MinMaxAggregateFunction final
   }
 
   Status Finalize(const StateType& src, Datum* output) const override {
-    *output = Datum({Datum(src.min), Datum(src.max)});
+    using ScalarType = typename TypeTraits<ArrowType>::ScalarType;
+    if (src.has_nulls) {
+      *output = Datum({Datum(std::make_shared<ScalarType>()), Datum(src.max)});
```

Comment:
Please ensure that both of these are null datums

Reply:
@bkietz so `skip_nulls` param will need to affect both min and max, will modify to use this logic.
during my tests, I found in arrow null scalar with all types are actually 0(https://github.com/apache/arrow/pull/5981/files#diff-382aa278b664f2f08a21632ed6b058ccR373-R377). Is it possible to return a null scalar = NAN for float type in arrow?

Reply:
Null scalar values (and values in a null array slot) are explicitly unspecified in the arrow protocol. If the output is null, you don't need to set the value to anything (since accessing it would be undefined behavior).

Reply:
I see, thanks for the explanation. Initially I was try to follow the skip_na option in pandas min max (https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.max.html) API, which seems to return "NAN" if skip_na = false